tests/Support/Constraint/CssConstraint.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Tests\Support\Constraint;
+
+/**
+ * Provides a method for use in constraints making assertions about CSS content where whitespace may vary.
+ *
+ * @author Jake Hotson <jake.github@qzdesign.co.uk>
+ */
+trait CssConstraint
+{
+    /**
+     * Processing of @media rules may involve removal of some unnecessary whitespace from the CSS placed in the <style>
+     * element added to the document, due to the way that certain parts are `trim`med.  Notably, whitespace either side
+     * of "{", "}" and "," or at the beginning of the CSS may be removed.
+     *
+     * This method helps takes care of that, by converting a search needle for an exact match into a regular expression
+     * that allows for such whitespace removal, so that the tests themselves do not need to be written less humanly
+     * readable and can use inputs containing extra whitespace.
+     *
+     * @param string $needle Needle that would be used with `assertContains` or `assertNotContains`.
+     *
+     * @return string Needle to use with `assertRegExp` or `assertNotRegExp` instead.
+     */
+    private static function getCssNeedleRegExp(string $needle): string
+    {
+        $needleMatcher = \preg_replace_callback(
+            '/\\s*+([{},])\\s*+|(^\\s++)|(>)\\s*+|(?:(?!\\s*+[{},]|^\\s)[^>])++/',
+            static function (array $matches): string {
```

Comment:
Please let's move this into a non-anonymous function with a proper description and a more specific parameter type annotation

Reply:
Oops.  Forgot about this.  Will address...

Reply:
Now done.